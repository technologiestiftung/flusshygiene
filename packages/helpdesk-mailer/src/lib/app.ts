import express from "express";
import cors, { CorsOptions } from "cors";
import morgan from "morgan";
import http from "http";
import helmet from "helmet";
import { router } from "./router";

const app = express();
const whitelist = [
  "https://www.flusshygiene.xyz",
  "https://www.flussbaden.org",
  "https://flussbaden.org",
];
if (process.env.NODE_ENV === "development") {
  const locals = [
    "http://localhost:3000",
    "http://localhost:8888",
    "http://localhost:5004",
  ];
  whitelist.push(...locals);
}
const corsOptions: CorsOptions = {
  origin: function (origin, callback) {
    if (!origin || whitelist.indexOf(origin) !== -1) {
      callback(null, true);
    } else {
      console.error(origin, "Not allowed by CORS");
      callback(new Error("Not allowed by CORS"));
    }
  },
};
app.use(cors());
app.use(express.json());
app.use(express.urlencoded({ extended: true }));
app.use(cors(corsOptions));
if (process.env.NODE_ENV === "development") {
  app.options("*", cors());
  app.use(morgan("combined"));
} else {
  app.use(helmet());
  app.use(morgan("combined"));
}

app.use("/helpdesk/v1", router);

const server = http.createServer(app);

export { server };
