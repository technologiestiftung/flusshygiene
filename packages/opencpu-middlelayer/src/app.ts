import { REDIS_HOST, REDIS_PORT, SESSION_SECRET } from './common/constants';

import cors from 'cors';
import express from 'express';
import helmet from 'helmet';
import { logStream, logger } from './logger';
import morgan from 'morgan';
import redis from 'redis';
import { router } from './http-router';
import session from 'express-session';
import uuidv4 from 'uuid/v4';
import { timeoutMiddleware, sleepFuncCheck } from './middleware';
import expressWs from 'express-ws';

const RedisStore = require('connect-redis')(session);
const client = redis.createClient({ host: REDIS_HOST, port: REDIS_PORT });
client.on('error', console.error);
client.on('connect', () => {
  logger.info('redis client has connected');
});

client.on('end', () => {
  logger.info('redis client session has disconnected');
});

const eapp = express();
const wsInstance = expressWs(eapp);
const { app } = wsInstance;
const wss = wsInstance.getWss();
// const ewss = expressWs(app);
// const wss = expressWs.getWss();
app.use(
  session({
    genid: function(_req) {
      return uuidv4(); // use UUIDs for session IDs
    },
    store: new RedisStore({ client }),
    secret:
      SESSION_SECRET !== undefined
        ? SESSION_SECRET
        : '45364f84-f39c-449d-b997-27f20da7e8ec',
    resave: false,
    saveUninitialized: true,
  }),
);
const origins = [process.env.APP_HOST_1!, process.env.APP_HOST_2!];
if (process.env.NODE_ENV === 'development') {
  origins.push('http://localhost');
}
app.use(
  cors({
    origin: origins,
  }),
);

if (process.env.NODE_ENV === 'development') {
  app.use(morgan('combined'));
} else if (process.env.NODE_ENV === 'test') {
} else {
  app.use(helmet());
  app.use(morgan('combined', { stream: logStream }));
}
app.use(express.json({ limit: '10mb' }));
app.use(express.urlencoded({ extended: true, limit: '10mb' }));
app.use(timeoutMiddleware);
app.use('/middlelayer', sleepFuncCheck, router);

app.ws('/middlelayer/socket', (ws, _req) => {
  // console.log(req.sessionID);
  // logger.info('req loc', req);
  ws.on('message', (data) => {
    // console.log(data);
    logger.info(`Message over ws ${data}`, data);
    // wss.clients.forEach((client: any) => {
    //   client.send(data);
    // });
  });
});

export { app, wss };
