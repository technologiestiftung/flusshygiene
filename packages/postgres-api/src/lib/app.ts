import cors, { CorsOptions } from 'cors';
// import errorHandler from 'errorhandler';
import express from 'express';
import helmet from 'helmet';
import morgan from 'morgan';
import { createConnection } from 'typeorm';
import routes from './routes';
import { CONSTANTS } from './common/constants';
import redis from 'redis';
import session from 'express-session';
import uuidv4 from 'uuid/v4';
import { cookieListing } from './middleware/cookie-listing';
import { logger } from './logger';
import publicRoutes from './routes-public';
import { errorHandler } from './middleware/errorHandler';

const RedisStore = require('connect-redis')(session);
const client = redis.createClient({
  host: CONSTANTS.REDIS_HOST,
  port: CONSTANTS.REDIS_PORT,
});
client.on('error', console.error);
client.on('connect', () => {
  logger.info('redis client has connected');
});

client.on('end', () => {
  logger.info('redis client session has disconnected');
});
const app = express();

try {
  createConnection()
    .then((_con) => {
      if (process.env.NODE_ENV === 'development') {
        process.stdout.write('connection established\n');
      }
    })
    .catch((err) => {
      console.error(err);
      throw err;
    });
} catch (error) {
  console.error(error);
}

const whitelist = [
  'https://www.flusshygiene.xyz',
  'http://localhost:3000',
  'http://localhost:8888',
];
const corsOptions: CorsOptions = {
  origin: function(origin, callback) {
    if (!origin || whitelist.indexOf(origin) !== -1) {
      callback(null, true);
    } else {
      callback(new Error('Not allowed by CORS'));
    }
  },
};

app.use(cors(corsOptions));
if (process.env.NODE_ENV === 'development') {
  app.options('*', cors());
  app.use(morgan('combined'));
} else {
  app.use(helmet());
  app.use(morgan('combined'));
}
app.use(express.json({ limit: '10mb' }));
app.use(express.urlencoded({ extended: true, limit: '10mb' }));

app.use('/api/v1/public', publicRoutes);
app.use(
  session({
    genid: function(_req: any) {
      return uuidv4(); // use UUIDs for session IDs
    },
    store: new RedisStore({ client }),
    secret:
      CONSTANTS.SESSION_SECRET !== undefined
        ? CONSTANTS.SESSION_SECRET
        : '45364f84-f39c-449d-b997-27f20da7e8ec',
    resave: false,
    saveUninitialized: true,
  }),
);

app.use(cookieListing);
app.use('/api/v1', routes);

// if (process.env.NODE_ENV === 'development') {
// In Express an error handler,
// always has to be the last line before starting the server.
// app.use(errorHandler());
app.use(errorHandler);
// } else {
// }

export = app;
