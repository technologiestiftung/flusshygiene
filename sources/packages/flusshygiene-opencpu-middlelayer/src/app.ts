import { SESSION_SECRET } from './common/constants';
import cors from 'cors';
import express from 'express';
import helmet from 'helmet';
import { logStream } from './logger';
import morgan from 'morgan';
import { router } from './http-router';
import session from 'express-session';

const app = express();
app.use(
  session({
    secret:
      SESSION_SECRET !== undefined
        ? SESSION_SECRET
        : '45364f84-f39c-449d-b997-27f20da7e8ec',
    resave: false,
    saveUninitialized: true,
  }),
);
app.use(cors());
if (process.env.NODE_ENV === 'development') {
  app.use(morgan('combined'));
} else if (process.env.NODE_ENV === 'test') {
} else {
  app.use(helmet());
  app.use(morgan('combined', { stream: logStream }));
}
app.use(express.json({ limit: '10mb' }));
app.use(express.urlencoded({ extended: true, limit: '10mb' }));
app.use('/', router);
export default app;
