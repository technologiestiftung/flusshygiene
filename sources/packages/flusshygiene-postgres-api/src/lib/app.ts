import cors from 'cors';
import errorHandler from 'errorhandler';
import express from 'express';
import helmet from 'helmet';
import morgan from 'morgan';
import routes from './routes';

const app = express();

app.use(cors());
if (process.env.NODE_ENV === 'development') {
  app.use(morgan('dev'));
} else {
  app.use(helmet());
  app.use(morgan('tiny'));
}
app.use(express.json());
app.use(express.urlencoded({extended: true}));
app.get('/', (request, response) => {
    response.send(`Server is running. You called ${request.url}`);
  });
app.use('/api/v1', routes);
// app.use('/api/v1', router);
if (process.env.NODE_ENV === 'development') {
  // In Express an error handler,
  // always has to be the last line before starting the server.
  app.use(errorHandler());
}
export = app;
