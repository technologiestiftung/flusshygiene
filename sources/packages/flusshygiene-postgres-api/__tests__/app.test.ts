
import request from 'supertest';
jest.mock('pg');
import app from '../src/lib/app';
// let rclient: redis.RedisClient;
jest.useFakeTimers();

beforeAll(() => {
});
afterAll(() => {
});
describe('default testing get requests', () => {
  test('should response with 200 on /', async () => {
    expect.assertions(2);
    const response = await request(app).get('/');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});
