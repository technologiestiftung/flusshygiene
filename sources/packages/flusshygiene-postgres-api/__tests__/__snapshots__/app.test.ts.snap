// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default testing get requests should response with 200 on / 1`] = `"Server is running. You called /"`;
